pkg/webhooks/machine_webhook.go

```diff
@@ -1332,6 +1335,58 @@ func validateMachineLifecycleHooks(m, oldM *machinev1.Machine) []error {
 	return errs
 }
 
+func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProviderSpec, parentPath *field.Path) []error {
+
+	var errs []error
+	dataDiskLuns := make(map[int32]struct{})
+	dataDiskNames := make(map[string]struct{})
+	reg := regexp.MustCompile(`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`)
+
+	for i, disk := range spec.DataDisks {
+		fldPath := parentPath.Index(i)
+
+		dataDiskName := machineName + "_" + disk.NameSuffix
+
+		if len(dataDiskName) > 80 {
+			errs = append(errs, field.Invalid(fldPath.Child("nameSuffix"), disk.NameSuffix, "too long, the overall disk name must not exceed 80 chars"))
+		}
+
+		if matched := reg.MatchString(disk.NameSuffix); !matched {
+			errs = append(errs, field.Invalid(fldPath.Child("nameSuffix"), disk.NameSuffix, "nameSuffix must be provided and can only contain letters, numbers, underscores, periods or hyphens"))
+		}
+
+		if _, exists := dataDiskNames[disk.NameSuffix]; exists {
+			errs = append(errs, field.Invalid(fldPath.Child("nameSuffix"), disk.NameSuffix, "each Data Disk must have a unique nameSuffix"))
+		}
+
+		if disk.DiskSizeGB < 4 {
+			errs = append(errs, field.Invalid(fldPath.Child("diskSizeGB"), disk.DiskSizeGB, "diskSizeGB must be provided and at least 4GB in size"))
+		}
+
+		if disk.Lun < 0 || disk.Lun > 63 {
+			errs = append(errs, field.Invalid(fldPath.Child("lun"), disk.Lun, "must be higher or equal than 0 and lower than 64"))
```

Comment:
Nit: Typically we would say `greater than or equal to 0`
```suggestion
			errs = append(errs, field.Invalid(fldPath.Child("lun"), disk.Lun, "must be greater than or equal to 0 and less than 64"))
```

Reply:
Yes, it definitely sounds better 👍 Updated.